src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -489,8 +489,39 @@ void MavlinkReceiver::handle_message_command_both(mavlink_message_t *msg, const
 			send_ack = true;
 		}
 
-	} else {
+	} else if (cmd_mavlink.command == MAV_CMD_DO_SET_ACTUATOR) {
+		// since we're only paying attention to 3 AUX outputs, the
+		// index should be 0, otherwise ignore the message
+		if (((int) vehicle_command.param7) == 0) {
+			actuator_controls_s actuator_controls{};
+			actuator_controls.timestamp = hrt_absolute_time();
+
+			// update with existing values to avoid changing unspecified controls
+			_actuator_controls_3_sub.update(&actuator_controls);
+
+			bool updated = false;
+
+			if (PX4_ISFINITE(vehicle_command.param1)) {
+				actuator_controls.control[5] = vehicle_command.param1;
+				updated = true;
+			}
+
+			if (PX4_ISFINITE(vehicle_command.param2)) {
+				actuator_controls.control[6] = vehicle_command.param2;
+				updated = true;
+			}
+
+			if (PX4_ISFINITE(vehicle_command.param3)) {
+				actuator_controls.control[7] = vehicle_command.param3;
+				updated = true;
+			}
+
+			if (updated) {
+				_actuator_controls_pubs[3].publish(actuator_controls);
```

Comment:
How about you publish control group 6 (might need to be defined) here, so it's conflict-free? https://docs.px4.io/master/en/concept/mixing.html#control-group-6-first-payload

Reply:
@LorenzMeier I think the rationale @bkueng had with using control group 3 was that it would work OOTB (mixers would not have to be modified by the user) and RC override is also possible. If group 6 is still preferable I can update

Reply:
@coderkalyan I was suggesting to handle the vehicle_command `MAV_CMD_DO_SET_ACTUATOR` directly in `rc_update.cpp` to avoid conflicts and then it would also just work for navigator.